.github/workflows/upload-test-results.yml

```diff
@@ -0,0 +1,93 @@
+          done
+      - name: Upload test results to S3
+        env:
+          S3_BUCKET: ${{ vars.TEST_RESULTS_BUCKET }}
+          AWS_ACCESS_KEY_ID: ${{ vars.TEST_RESULTS_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.TEST_RESULTS_AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: us-east-2
+        if: env.S3_BUCKET != '' && env.AWS_ACCESS_KEY_ID != '' && env.AWS_SECRET_ACCESS_KEY != ''
+        shell: bash --noprofile --norc -euo pipefail {0}
+        run: |
+          # 1. Don't prefix attributes, because +@ (the default prefix) is not a valid character in nested row field names in the Hive connector for JSON files.
+          # 2. When converting to JSON, make sure 'testcase' is always an array: https://mikefarah.gitbook.io/yq/usage/xml#parse-xml-force-as-an-array
+          # 3. Remove system-err and system-out, because they cannot be easily parsed and add significant bloat, making storing and processing the data much more costly.
+          # 4. Remove properties, because they leak secret values.
+          # 5. Truncate all strings to 1k characters to avoid having lines longer than 100MB.
+          yq_opts=(
+              --input-format=xml
+              --output-format=json
+              --xml-attribute-prefix=''
+              --xml-content-name='content'
+              --xml-skip-directives
+              --xml-skip-proc-inst
+              '.testsuite.testcase |= ([] + .) | .testsuite.testcase[] |= del(.system-err, .system-out) | .testsuite |= del(.properties) | .. |= select(tag == "!!str") |= sub("(.{0,1000}).*", "${1}")'
+          )
+          artifact_id='${{ github.event.workflow_run.id }}-${{ github.event.workflow_run.run_attempt }}.json.gz'
+
+          find . \
+            -name TEST-\*.xml \
+            -exec yq "${yq_opts[@]}" {} \; \
+                | jq -c > surefire.ndjson
+          find . \
+            -name testng-results.xml \
+            -exec yq "${yq_opts[@]}" {} \; \
+                | jq -c > testng.ndjson
+
+          for filename in *.ndjson; do
+            if [ ! -s "$filename" ]; then
+              continue;
+            fi
+            jq -c \
+              --argjson addObj '{"branch":"${{ github.event.workflow_run.head_branch }}","git_sha":"${{ github.event.workflow_run.head_sha }}","workflow_name":"${{ github.event.workflow.name }}","workflow_run":"${{ github.event.workflow_run.id }}","workflow_conclusion":"${{ github.event.workflow_run.conclusion }}","workflow_job":"","workflow_run_attempt":"${{ github.event.workflow_run.run_attempt }}","timestamp":""}' \
+              --arg timestamp "$(date -u '+%F %T.%3NZ')" \
+              '. + $addObj | .timestamp=$timestamp' "$filename" | gzip -c > "$artifact_id"
+
+            aws s3 cp --no-progress "$artifact_id" "s3://$S3_BUCKET/tests/results/type=$(basename "$filename" .ndjson)/owner=${{ github.repository_owner }}/repo=$(basename "${{ github.repository }}")/date_created=$(date -u '+%Y-%m-%d')/$artifact_id"
```

Comment:
Why do we need to namespaces by "owner" (I assume this is trinodb?). Everything in this bucket should be for trinodb (and only for it)

Reply:
This is for any forks, so they don't need changes

Reply:
Forks will not be able to upload to that bucket anyway, since they don’t have the appropriate keys and permissions.

Reply:
The bucket is configurable. Upload won't be attempted if it's not set.

Reply:
Also note it made testing easier. I ran this in my fork. Let me know if you have strong feelings about this.

Reply:
> Also note it made testing easier. I ran this in my fork. Let me know if you have strong feelings about this.

That only worked because you know the keys to write to that bucket. But it doesn't generalize.

I would get rid of that part. It doesn't seem to provide any value and is redundant with the bucket itself, which is dedicated to trinodb already.

Reply:
Removed